src/wordy/ast/BinaryExpressionNode.java

```diff
@@ -59,4 +62,70 @@ public String toString() {
     protected String describeAttributes() {
         return "(operator=" + operator + ')';
     }
+
+    @Override
+    protected double doEvaluate(EvaluationContext context) {
```

Comment:
In this function as well as the compile function for BinaryExpression, I used a lot of if/else statements. Paul's solution has something called switch, which I've definitely seen/heard of, but never used. I was doing some reading online about peoples' opinions on switch to try and understand the difference. One person from stack exchange was saying that the difference is not functionality, but intent. They essentially said if you have something only being impacted by one variable (like operator), that a switch statement can be a lot better. If you use a series of ifs, someone working on your project in the future may not realize that it's only one variable being dealt with and add on random if-statements that don't have to do with the one variable, and then that makes the code become more confusing and less well-organized. To me, that explanation makes a lot of sense.